Building dynamic library: undefined references (Linux)

Here's the gist with a simple project: https://gist.github.com/nigredo-tori/9f774894c38f7a6a607270a6a456ea20 (run `test.sh` to reproduce the issue)
I build a Nim file as a dynamic library, and use it from the C file. However, the linker fails:

```
./libfoo.so: undefined reference to `cmdCount'
./libfoo.so: undefined reference to `cmdLine'
collect2: error: ld returned 1 exit status
```

This seems to be triggered by importing the `logging` module. Emitting code that defines the corresponding C variables allows to work around the issue.

--- a/lib/pure/os.nim
+++ b/lib/pure/os.nim
@@ -1427,7 +1427,7 @@ elif defined(windows):
     if isNil(ownArgv): ownArgv = parseCmdLine($getCommandLine())
     return TaintedString(ownArgv[i])
 
-elif not defined(createNimRtl):
+elif not defined(createNimRtl) and not(defined(posix) and appType == "lib"):
   # On Posix, there is no portable way to get the command line from a DLL.
   var
     cmdCount {.importc: "cmdCount".}: cint

